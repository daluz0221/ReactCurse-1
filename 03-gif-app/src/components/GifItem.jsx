import React from 'react';
import PropTypes from 'prop-types';


const GifItem = ({id, title, url}) => {

    // console.log(img);
    return (
        <div className="card animate__animated animate__fadeInUp">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifItem;
