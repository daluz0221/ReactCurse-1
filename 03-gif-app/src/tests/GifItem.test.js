import '@testing-library/jest-dom';
import {shallow} from 'enzyme'
// import GifGrid from '../components/GifGrid';
import GifItem from '../components/GifItem';

describe('Pruebas en el componente <GifItem />', () => {
    
    
    const title = 'un titulo';
    const url = 'https://localhost/mialgo.jpg';
    const wrapper = shallow(<GifItem title={title} url={url} />);
   

    

    test('Debe crear el componente y contrastarlo con el snapshot', () => {

        expect(wrapper).toMatchSnapshot();

    });
    
    
    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(title);
    });

    test('probando los props en la img', () => {
       
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('debo probar la clase del div animate__fadeinup', () => {
       
        const div = wrapper.find('div');

        expect(div.prop('className').split(' ')[2]).toBe('animate__fadeInUp');

    });
    
    
    

});
