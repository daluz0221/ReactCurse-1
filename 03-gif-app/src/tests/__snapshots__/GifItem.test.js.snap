// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem /> Debe crear el componente y contrastarlo con el snapshot 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="un titulo"
    src="https://localhost/mialgo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
